src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1160,6 +1161,55 @@ ss::future<end_tx_reply> tx_gateway_frontend::end_txn(
       });
 }
 
+ss::future<tx_gateway_frontend::return_all_txs_res>
+tx_gateway_frontend::get_all_transactions() {
+    auto shard = _shard_table.local().shard_for(model::tx_manager_ntp);
+
+    if (shard == std::nullopt) {
+        vlog(txlog.warn, "can't find a shard for {}", model::tx_manager_ntp);
+        co_return tx_errc::shard_not_found;
+    }
+
+    co_return co_await container().invoke_on(
+      *shard,
+      _ssg,
+      [](tx_gateway_frontend& self)
+        -> ss::future<tx_gateway_frontend::return_all_txs_res> {
+          auto partition = self._partition_manager.local().get(
+            model::tx_manager_ntp);
+          if (!partition) {
+              vlog(
+                txlog.warn,
+                "can't get partition by {} ntp",
+                model::tx_manager_ntp);
+              co_return tx_errc::invalid_txn_state;
```

Comment:
It is popular pattern in `tx_gateway_frontend`. Maybe invalid_txn_state is not right error, because @rystsov used it for the same problem in changing transaction state

Reply:
@rystsov should chime in here I don't know the answer.

copying the pattern is completely reasonable, but it also means that code gets reviewed again. and in this case it just seemed that it wasn't an invalid state but rather a unlikely-to-occur-but-valid state.

Reply:
When a partition isn't found we return `partition_not_found` from `init_tm_tx` and `invalid_txn_state` from `add_partition_to_tx`, `add_offsets_to_tx` and from `end_txn`. The latter methods anyway fail when leadership changes so we just make them fail faster.

In this case we should return something like `partition_not_found` or `not_coordinator`